### PR TITLE
added MDM functionalities

### PR DESCRIPTION
**Pull request by shenda1**
# Description
PR is raised to add MDM functionalities

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
https://github.com/dell/csm/issues/778

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [ ] Integration Tests
- [ ] Unit Tests

image

### CODE REVIEW

**Comment by shenda1** [comment]
image

File: system.go [hunk 1]
```diff
@@ -106,3 +106,96 @@ func (s *System) CreateSnapshotConsistencyGroup(
 
 	return &snapResp, nil
 }
+
+// GetMDMClusterDetails returns MDM cluster details
+func (s *System) GetMDMClusterDetails() (*types.MdmCluster, error) {
```
Comment (khareRajshree) [inline system.go:111]: Hi @shenda1, does these functionality work with powerflex system v4.0 as well? Can you share some results where the details for 3,6 MDMs and 4.0 MDMs are retrieved.
Thanks.

File: system.go [hunk 2]
```diff
@@ -106,3 +106,96 @@ func (s *System) CreateSnapshotConsistencyGroup(
...
+	err := s.client.getJSONWithRetry(
+		http.MethodPost, path, switchClusterMode, nil)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// ChangeMdmOwnerShip modifies the primary MDM
+func (s *System) ChangeMdmOwnerShip(id string) error {
```
Comment (khareRajshree) [inline system.go:188]: Can we add unit tests as well for the new methods introduced. Thanks.

**Review by khareRajshree — CHANGES_REQUESTED** [review]

File: system.go (hunk 1 above)
Comment (shenda1) [reply system.go:111]: Hi @khareRajshree The changes are tested only w.r.t PowerFlex version 3.6 as per the Terraform support matrix.

File: system.go (hunk 2 above)
Comment (shenda1) [reply system.go:188]: Hi @khareRajshree Unit tests are added

**Review by khareRajshree — APPROVED** [review]

**Review by ashleyvjoy — APPROVED** [review]

**Pull request merged**